physics/discrete_trajectory2.hpp

```diff
@@ -41,35 +47,31 @@ class DiscreteTrajectory2 : public Trajectory<Frame> {
   DiscreteTrajectory2(const DiscreteTrajectory2&) = delete;
   DiscreteTrajectory2& operator=(const DiscreteTrajectory2&) = delete;
 
-    Iterator begin() const;
-  Iterator end() const;
+  iterator begin() const;
+  iterator end() const;
 
-  Iterator rbegin() const;
-  Iterator rend() const;
+  reverse_iterator rbegin() const;
+  reverse_iterator rend() const;
 
-  Iterator find(Instant const& t) const;
+  iterator find(Instant const& t) const;
 
-  Iterator lower_bound(Instant const& t) const;
-  Iterator upper_bound(Instant const& t) const;
+  iterator lower_bound(Instant const& t) const;
+  iterator upper_bound(Instant const& t) const;
 
-  SegmentIterator segments_begin() const;
-  SegmentIterator segments_end() const;
-
-  SegmentIterator segments_rbegin() const;
-  SegmentIterator segments_rend() const;
+  SegmentRange segments() const;
+  ReverseSegmentRange rsegments() const;
```

Comment:
I think the Right Thing to do is to have only `segments` and use an adapter to iterate backward (in C++20 that would be https://en.cppreference.com/w/cpp/ranges/reverse_view, maybe absl has a thing in the meantime, or we can write one in `base`, we should have `make_reverse_iterator`).

Reply:
Added a TODO to fix this once we have C++20.